CMakeLists.txt

```diff
@@ -90,12 +90,14 @@ endif()
 # User-defined options.
 
 option (NN_STATIC_LIB "Build static library instead of shared library." OFF)
-option (NN_ENABLE_DOC "Enable building documentation." ON)
 option (NN_ENABLE_GETADDRINFO_A "Enable/disable use of getaddrinfo_a in place of getaddrinfo." ON)
 option (NN_TESTS "Build and run nanomsg tests" ON)
 option (NN_TOOLS "Build nanomsg tools" ON)
 option (NN_ENABLE_NANOCAT "Enable building nanocat utility." ${NN_TOOLS})
 
+set (NN_ENABLE_DOC "DEFAULT" CACHE STRING "Enable building documentation.")
+set_property (CACHE NN_ENABLE_DOC PROPERTY STRINGS "ON" "OFF" "DEFAULT")
```

Comment:
I'd only treat the positive cases there, and let the negative cases fall through to false if *anything* else (i.e. it doesn't concretely match one of the positive values)